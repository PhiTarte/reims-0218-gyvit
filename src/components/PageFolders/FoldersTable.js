import React from "react"
import { Table, Container } from "reactstrap"
import PropTypes from "prop-types"
import FoldersTableTbody from "./FoldersTableTbody"
import FoldersTableCategory from "./FoldersTableCategory"

const FoldersTable = ({
  files = [],
  dirs = [],
  parent = {},
  onDirclick,
  onBackclick
}) => (
  <div>
    <Container
      style={{
        padding: "15px",
        width: "100%",
        height: "650px",
        overflow: " auto"
      }}
    >
      <Table borderless striped>
        <thead>
          <tr>
            <th>
              <FoldersTableCategory name="Name  " direction={null} />
            </th>
            <th>
              <FoldersTableCategory name="Last update  " direction={null} />
            </th>
            <th>
              <FoldersTableCategory name="Users  " direction={"asc"} />
            </th>
            <th />
          </tr>
        </thead>
        <FoldersTableTbody
          onDirclick={onDirclick}
          onBackclick={onBackclick}
          files={files}
          parent={parent}
          dirs={dirs}
        />
      </Table>
    </Container>
  </div>
)

FoldersTable.propTypes = {
  files: PropTypes.array,
  dir: PropTypes.array
}

export default FoldersTable
