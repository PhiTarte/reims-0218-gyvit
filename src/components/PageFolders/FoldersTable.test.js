import React from "react"
import ReactDOM from "react-dom"
import renderer from "react-test-renderer"
import FoldersTable from "./FoldersTable"

it("renders without crashing", () => {
  const div = document.createElement("div")
  ReactDOM.render(<FoldersTable />, div)
  ReactDOM.unmountComponentAtNode(div)
})

it("renders link html", () => {
  const tree = renderer.create(<FoldersTable />).toJSON()
  expect(tree).toMatchSnapshot()
})
