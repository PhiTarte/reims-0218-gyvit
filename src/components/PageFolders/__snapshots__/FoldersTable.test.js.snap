// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders link html 1`] = `
<div>
  <div
    className="container"
  >
    <table
      className="table"
    >
      <thead>
        <tr>
          <th />
          <th>
            <div>
              Name
              <img
                alt="upDown"
                src="Assets/icon_double_arrow.png"
              />
            </div>
          </th>
          <th>
            <div>
              Last update
              <img
                alt="upDown"
                src="Assets/icon_double_arrow.png"
              />
            </div>
          </th>
          <th>
            <div>
              Users
              <img
                alt="down"
                src="Assets/icon_arrow_down.png"
              />
            </div>
          </th>
          <th />
        </tr>
      </thead>
      <tbody />
    </table>
  </div>
</div>
`;
